include/crab/domains/mru_region_domain.hpp

```diff
@@ -0,0 +1,1336 @@
+    // project cache_lines domain
+    m_cache_lines.project(base_vars);
+    // project regs domain
+    m_regs.project(base_vars);
+    // project addrs domain
+    m_addrs.project(base_vars);
+    // project flags domain
+    m_flags.project(base_vars);
+
+    // -- update m_var_map and m_rev_var_map
+    std::vector<variable_t> var_map_to_remove;
+    var_map_to_remove.reserve(m_var_map.size());
+    for (auto &kv : m_var_map) {
+      if (!std::binary_search(sorted_variables.begin(), sorted_variables.end(),
+                              kv.first)) {
+        var_map_to_remove.push_back(kv.first);
+        kv.second.remove_rev_varmap(m_rev_var_map);
+      }
+    }
+    for (auto &v : var_map_to_remove) {
+      m_var_map.erase(v);
+    }
+  }
+
+  std::string domain_name() const override {
+    return "CacheDomain(" + m_cache_lines.domain_name() + ", " +
+           m_regs.domain_name() + ")";
+  }
+
+  void write(crab_os &o) const override {
+    if (is_bottom()) {
+      o << "_|_";
+    } else if (is_top()) {
+      o << "{}";
+    } else {
+      CRAB_LOG("mru-region-print", o << "(CacheLine=" << m_cache_lines << ",\n"
+                                     << "Regs=" << m_regs << ",\n"
+                                     << "Addrs=" << m_addrs << ",\n"
+                                     << "Flags=" << m_flags << ")\n";);
+
+      std::unordered_map<std::string, std::string> renaming_map;
+      auto get_type_fn = [](const variable_t &v) -> variable_type {
+        return variable_type(variable_type_kind::UNK_TYPE);
+      };
+      // Assigns a string name to each ghost variable
+      ghost_variables_t::mk_renaming_map(m_rev_var_map, get_type_fn,
+                                         renaming_map);
+      m_alloc.add_renaming_map(renaming_map);
+      o << "(CacheLine=" << m_cache_lines << ",\n"
+        << "Regs=" << m_regs << ",\n"
+        << "Addrs=" << m_addrs << ",\n"
+        << "Flags=" << m_flags << ")\n";
+      m_alloc.clear_renaming_map();
+    }
+  }
+  /**------------- End domain API definitions -------------------**/
+
+  /**------------- Begin MRU API definitions -------------------**/
+  // perform a similiar work for cache.lines.constraint(cache.regs)
+  // This one requires to be renaming because mem base dom used different base
+  // vars
+  mru_region_domain_t cstrs() {
+    auto csts = m_regs.to_linear_constraint_system();
+    mru_region_domain_t res(*this);
+    res.m_cache_lines += csts;
+    return res;
+  }
+
+  // get all used Crab IR vars by region dom
+  void vars(variable_vector_t &used_vars) const {
+    used_vars.reserve(m_var_map.size());
+    for (auto &kv : m_var_map) {
+      used_vars.push_back(kv.first);
+    }
+  }
+
+  void region_vars(variable_vector_t &used_vars) const {
+    used_vars.reserve(m_var_map.size());
+    for (auto &kv : m_var_map) {
+      if (kv.first.get_type().is_region()) {
+        used_vars.push_back(kv.first);
+      }
+    }
+  }
+
+  // NOTE: For debug use
```

Comment:
these methods for debugging should be private. Remember to add `CRAB_LOG` to print extra stuff.

Reply:
Is that possible to pass some flags for `test-bin`? I used `test-bin/fold1`, `test-bin/expand1` for unit testing. I know we can pass them through the `clam.py` or `crab` executable.